examples/demo13.py

```diff
@@ -0,0 +1,378 @@
+# Copyright (c) 2012-2014 by the GalSim developers team on GitHub
+# https://github.com/GalSim-developers
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+"""
+Demo #13
+
+The thirteenth script in our tutorial about using Galsim in python scripts: examples/demo*.py.
+(This file is designed to be viewed in a window 100 characters wide.)
+
+This script currently doesn't have an equivalent demo*.yaml or demo*.json file.
+
+This script introduces non-idealities arising from NIR detectors, in particular those that will be
+observed and accounted for in the WFIRST surveys. Three such non-ideal effects are demonstrated, in
+the order in which they are introduced in the detectors:
+
+1) Reciprocity failure: Flux-dependent sensitivity of the detector.
+2) Non-linearity: Charge-dependent gain in converting from units of electrons to ADU.  Non-linearity
+   in some form is also relevant for CCDs in addition to NIR detectors.
+3) Interpixel capacitance: Influence of charge in a pixel on the voltage reading of neighboring
+   ones.
+
+The purpose of the demo is two-fold: (1) to show the effects of detector non-idealities on images
+from NIR detectors, and (2) to illustrate the full image generation process, including all sources
+of noise at appropriate stages.
+
+New features introduced in this demo:
+- galsim.makeCOSMOSCatalog(...)
+- galsim.makeCOSMOSObj(...)
+- Adding sky level and dark current.
+- image.addReciprocityFailure(exp_time, alpha, base_flux)
+- image.quantize()
+- image.applyNonlinearity(NLfunc, *args)
+- image.applyIPC(IPC_kernel, edge_treatment, fill_value, kernel_nonnegativity,
+                 kernel_normalization)
+- readnoise.setReadNoise(readnoise_level)
+- All functionality in the galsim.wfirst module.
```

Comment:
Should list these by name, since undoubtedly more `galsim.wfirst` features will be added at some point, and not all of them will necessarily be added to this demo.